packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx

```diff
@@ -1,14 +1,17 @@
 import { BridgeDetails } from "../../components/bridge/BridgeDetails.tsx";
 import { Link, useParams } from "react-router-dom";
 import { useBridge } from "../../hooks/data/bridges.ts";
-import { useCallback, useEffect, useState } from "react";
+import { memo, useCallback, useEffect, useState } from "react";
 import { useNotifications } from "@toolpad/core";
 import { useDevices } from "../../hooks/data/devices.ts";
 import { DeviceList } from "../../components/device/DeviceList.tsx";
 import { useTimer } from "../../hooks/timer.ts";
 import { IconButton, Stack, Typography } from "@mui/material";
 import Box from "@mui/material/Box";
-import { Edit } from "@mui/icons-material";
+import { Edit, Refresh } from "@mui/icons-material";
+
+const MemoizedBridgeDetails = memo(BridgeDetails);
+const MemoizedDeviceList = memo(DeviceList);
```

Comment:
`memo` prevents re-rendering if NO parameter changes. Object equality is checked by checking the object reference and not its content. When refreshed the device objects are whole new objects from the response. Even if the content is the same, it’s still a new object. Memo won’t have any effect.

Reply:
From my debugging I found out that without the `memo`, each time `setTimer` is invoked it triggers a re-rendering of the whole device list, whereas with `memo` the device list will only be re-rendered when its props (`devices`) change.

Reply:
interesting. let's keep it like this, since it doesnt hurt.